Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -1165,7 +1164,20 @@ bool RenderLayerCompositor::updateCompositingLayers(CompositingUpdateType update
     return true;
 }
 
-void RenderLayerCompositor::computeCompositingRequirements(RenderLayer* ancestorLayer, RenderLayer& layer, LayerOverlapMap& overlapMap, CompositingState& compositingState, BackingSharingState& backingSharingState, bool& descendantHas3DTransform)
+// Unchanged leaf compositing layers that clip their descendants can skip descendant
+// traversal, since their descendants can't contribute any new overlap to the map.
+static bool canSkipComputeCompositingRequirementsForSubtree(RenderLayer& layer, bool willBeComposited)
+{
+    if (layer.needsCompositingRequirementsTraversal() || layer.hasDescendantNeedingCompositingRequirementsTraversal())
+        return false;
+
+    if (!layer.isComposited() || !willBeComposited || layer.hasCompositingDescendant())
+        return false;
+
+    return layer.hasCompositedScrollableOverflow();
```

Comment:
This could potentially be generalised to some sort of 'the paint-order descendants of this layer don't add new overlap'.

We could accumulate a bool when we do traverse the subtree, checking if the overlap map got mutated usefully, and then use that to know for future passes.

Reply:
I'd rather a check for stacking context + non-visible overflow. We want paint containment to trigger this optimization.

Reply:
Non-visible overflow is definitely a good change, will do.

Happy to add stacking context too, but I'm struggling to convince myself that it's needed :)